src/Microsoft.Diagnostics.Monitoring.WebApi/EventMonitoringPassthroughStream.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    /// <summary>
+    /// A stream that can monitor an event stream which is compatible with <see cref="EventPipeEventSource"/>
+    /// for a specific event while also passing along the event data to a destination stream.
+    /// </summary>
+    public class EventMonitoringPassthroughStream : Stream
```

Comment:
If we were to migrate this code, is EventMonitoringPassthroughStream & StreamLeaveOpenWrapper sufficient?